### PR TITLE
Optimized XmlCommentsOperationFilter logic

### PR DESCRIPTION
**Pull request by xperiandri**
Optimized `XmlCommentsOperationFilter.ApplyResponsesXmlToResponses` and `XmlCommentsOperationFilter.ApplyPropertiesXmlToPropertyParameters` search code.

Nothing changed in behavior

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs [hunk 1]
```diff
@@ -98,7 +99,13 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
             {
                 // Check for a corresponding  API parameter (from ApiExplorer) that's property-bound?
                 var propertyParam = apiDescription.ParameterDescriptions
-                    .Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata?.PropertyName != null)
```
Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: @Bjego, here double check is done that can be eliminated.
If you check for null than do something once `?.` is nice but if you apply `?.` to the same property more than once it's a waste of resources.
This short form does not add any value in readability too.

Comment (Bjego) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: Ok yes I think your right ;) I didn`t think about the resources.

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs [hunk 2]
```diff
@@ -98,7 +99,13 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
...
+                    .Where(p =>
+                    {
+                        var modelMetadata = p.ModelMetadata;
+                        return modelMetadata != null
+                            && modelMetadata.ContainerType != null
+                            && modelMetadata.PropertyName != null;
+                    })
```
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: what's the advantage of this change vs just removing the second (redundant use of ?)? In other words, why not just replace:

    Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata?.PropertyName != null)

with

    Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata.PropertyName != null)

Is it just a style preference? If so, I still prefer the one-liner.

Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: Nothing about style, only performance.

Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: If you prefer the second one, that's fine too.
My option is more readable in my opinion.

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs [hunk 3]
```diff
@@ -80,9 +80,10 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
             while (responseNodes.MoveNext())
             {
                 var code = responseNodes.Current.GetAttribute("code", "");
-                var response = responses.ContainsKey(code)
-                    ? responses[code]
-                    : responses[code] = new Response();
+
+                Response response;
+                if (!responses.TryGetValue(code, out response))
+                    responses[code] = response = new Response();
```
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: I like this change

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs (hunk 2 above)
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: I don't like this change and would prefer to stick with the following:

    Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata.PropertyName != null)

Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: So we're hitting subjective discussion about readability now. Like I said, I prefer the one-liner.

Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: @xperiandri - do you want to update as discussed and then I'll be happy to merge. Thanks again for the contributions.

Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: OK

**Pull request closed without merging**